### DISCUSSION TITLE
List of all font names

### DISCUSSION THREAD

**Discussion by EmanueleGusso** [jsvine/pdfplumber#458] [Q&A] [answered]
Hi everybody,
first of all thank you and congratulations for this wonderful work.
The big question is: is there a list of all possible font names that pdplumber can recognize? It would be extremely useful for my purpose.
Thank you very much in advance!

**Comment by samkit-jain** [comment, marked as answer]
Hi @EmanueleGusso Appreciate your interest in the library. You may use the following code to get all the fontnames in a PDF
```python
import pdfplumber

pdf = pdfplumber.open("my_file.pdf")

print(set(char["fontname"] for char in pdf.chars))
```

There's no direct method available to fetch the fontnames so you'll have to go through all the characters and then fetch the fontnames from them using the `fontname` property in a char object.

**Reply by EmanueleGusso** [reply]
Thank you very much for your prompt response! This will be definitely useful for my work.
So you mean that there is not a list of all possible (not just in a specific pdf file) fontnames that pdfplumber can recognize?

**Reply by samkit-jain** [reply]
I am not sure if there exists a list of all the font names that `pdfminer.six` (library on which pdfplumber depends) can recognise.

**Reply by samkit-jain** [reply]
Because there can be hundreds of new fonts that are being developed every day and the library just reads whatever is readable.
